umm-spec-lib/src/cmr/umm_spec/validation/platform.clj

```diff
@@ -34,24 +34,43 @@
    with the model number. So the short name is not specific enough to identify
    specific platform instances. Therefore we are adding Characteristic names and values
    so that specific aircraft tail numbers, or boat IDS, etc. can be specified with the
-   short name."
+   short name. The |-| string is also added as a delimiter between the fields. This is used
+   in the getErrorString parameter-parser to parse out the different fields, so that a
+   meaningful error string can be suppplied back to the user."
   [values unique-fields]
   (let [{:keys [ShortName Characteristics Name Value]} unique-fields]
     (map (fn [value]
            (apply str
                  (ShortName value)
-                 (map #(str (Name %) (Value %)) (Characteristics value))))
+                 "|-|"
+                 (map #(str (Name %) "|-|" (Value %)) (Characteristics value))))
          values)))
 
+(defn- getErrorString
+  "Creates the error string for platforms depending on which elements (ShortName, Characteristic/Name, Characteristic/Value) are present.
+   The element order is ShortName, Name, Value. The values are separated by the |-| string."
+  [duplicate-name]
+  (let [platformElements (clojure.string/split duplicate-name #"\|-\|")
+        platCounts (count platformElements)]
+    ;; The case of 0 or greater than 3 shouldn't exist.  If it is either value then the validation
+    ;; software is not working.
+    (if (or (< platCounts 1) (> platCounts 3)) (throw (Exception. "The platform validation checking for duplicates either has 0 or more than 3 elements, neither is valid. There is a software problem."))
+      (case platCounts
+        1  (format "The Platform ShortName [%s] must be unique. This record contains duplicates." (get platformElements 0))
+        2  (format "The combination of Platform ShortName [%s] along with its Characteristic Name [%s] must be unique. This record contains duplicates." (get platformElements 0) (get platformElements 1))
```

Comment:
I don't see a test case that makes use of this error msg.  Could you add one, if it doesn't exist?